python/cudf/cudf/core/series.py

```diff
@@ -5201,14 +5196,47 @@ def value_counts(
                 index=cudf.Index([], dtype=self.dtype),
             )
 
-        res = self.groupby(self, dropna=dropna).count(dropna=dropna)
+        if bins is not None:
+            res = self.groupby(series_bins, dropna=dropna).count(dropna=dropna)
+            res = res[res.index.notna()]
+            res = res.sort_index()
+        else:
+            res = self.groupby(self, dropna=dropna).count(dropna=dropna)
+
         res.index.name = None
 
         if sort:
             res = res.sort_values(ascending=ascending)
 
         if normalize:
             res = res / float(res._column.sum())
+
+        # Pandas returns an IntervalIndex as the index of res
+        # this condition makes sure we do too if bins is given
+        if bins is not None and len(res) == len(res.index.categories):
+            # Thinking through a less terrible way to change Categorical
+            # index to an IntervalIndex
+            left_interval = res.index.dtype.categories._values.children[0]
```

Comment:
Just mentioning that I'm aware this section of code is not great. The issue is that from `cudf.cut` we return a `CategoricalIndex` while pandas returns an `IntervalIndex`.  This code is a bit brute force but works. I'm working on creating a better method for this change, but for the moment if you have any ideas feel free to share. Thanks!